tests/e2e/test_small_file_workload.py

```diff
@@ -0,0 +1,75 @@
+"""
+Test to exercise Small File Workload
+"""
+import logging
+import pytest
+import time
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility import templating
+from ocs_ci.ocs.utils import get_pod_name
+from ocs_ci.ocs.container import ContainerStor
+from ocs_ci.ocs.ripsaw import RipSaw
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import E2ETest, tier1
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='class')
+def ripsaw(request):
+    # Create Ceph Block Pool
+    sc = ContainerStor(sc_name='ceph-backed', cleanup=True)
+    # Create RipSaw Operator
+    ripsaw = RipSaw()
+    sc.create()
+
+    def teardown():
+        ripsaw.cleanup()
+        sc.delete()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
+@tier1
+class TestSmallFileWorkload(E2ETest):
+    """
+    Deploy Ripsaw operator and run SmallFile workload
+    """
+    def test_smallfile_workload(self, ripsaw):
+        """
+        Run SmallFile Workload
+        """
+        log.info("Apply Operator CRD")
+        ripsaw.apply_crd('resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml')
+
+        log.info("Running SmallFile bench")
+        sf_data = templating.load_yaml_to_dict(constants.SMALLFILE_BENCHMARK_YAML)
+        sf_obj = OCS(**sf_data)
+        sf_obj.create()
+        # wait for benchmark pods to get created - takes a while
+        time.sleep(20)
```

Comment:
please don't use sleep without sampling the status you are waiting for.
you have several functions to sample pods status

Reply:
At this stage, we dont have any objects to sample, based on the config, it will take a bit for those client pods to get created with custom names. Once created, the pod is being sampled later in the code.

Reply:
I would suggest using the function below "get_pod_name('smallfile-client', 'my-ripsaw')" with retry until you have the name.
What will happen if the pod will be there after 1 second? 19 seconds will be lost for nothing and in the future when we'll have hundreds of tests it will be critical

Reply:
Fixed